pandas/_typing.py

```diff
@@ -124,10 +124,8 @@
 Axes = Collection[Any]
 
 # dtypes
-NpDtype = Union[str, np.dtype]
-Dtype = Union[
-    "ExtensionDtype", NpDtype, type_t[Union[str, float, int, complex, bool, object]]
-]
+NpDtype = Union[str, np.dtype, type_t[Union[str, float, int, complex, bool, object]]]
+Dtype = Union["ExtensionDtype", NpDtype]
```

Comment:
does this PR need to go in before #41203?

Reply:
No, they should be independent.  Either can go first, and once one goes in, I can do the mypy check on the other.

Reply:
Would it make sense to use `NpDtype = npt.DTypeLike` to be consistent between numpy and pandas?

Reply:
> Would it make sense to use `NpDtype = npt.DTypeLike` to be consistent between numpy and pandas?

That's a big change, as it affects a lot more than just the `astype()` signature.  The types are different:
```python
>>> import numpy.typing as npt
>>> from pandas._typing import NpDtype
>>> NpDtype
typing.Union[str, numpy.dtype, typing.Type[typing.Union[str, float, int, complex, bool, object]]]
>>> npt.DTypeLike
typing.Union[numpy.dtype, NoneType, type, numpy.typing._dtype_like._SupportsDType[numpy.dtype], str, typing.Tuple[typing.Any, int], typing.Tuple[typing.Any, typing.Union[typing.SupportsIndex, typing.Sequence[typing.SupportsIndex]]], typing.List[typing.Any], numpy.typing._dtype_like._DTypeDict, typing.Tuple[typing.Any, typing.Any]]
```